client/hosting/sites/components/sites-dataviews/actions.tsx

```diff
@@ -0,0 +1,236 @@
+						site.ID,
+						WPCOM_FEATURES_SITE_PREVIEW_LINKS
+					);
+
+					if ( ! hasSitePreviewLinksFeature ) {
+						return false;
+					} */
+
+					return !! site.is_coming_soon;
+				},
+			},
+
+			{
+				id: 'copy-site',
+				label: __( 'Copy site' ),
+				callback: ( sites ) => {
+					const site = sites[ 0 ];
+					window.location.href = addQueryArgs( `/setup/copy-site`, {
+						sourceSlug: site.slug,
+					} );
+					dispatch( recordTracksEvent( 'calypso_sites_dashboard_site_action_copy_site_click' ) );
+				},
+				isEligible: ( site ) => {
+					const isWpcomStagingSite = site.is_wpcom_staging_site;
+
+					/* const shouldShowSiteCopyItem = useSafeSiteHasFeature(
+						site?.ID,
+						WPCOM_FEATURES_COPY_SITE
+					);
+					return ! isWpcomStagingSite && shouldShowSiteCopyItem; */
+					return ! isWpcomStagingSite;
+				},
+			},
+
+			{
+				id: 'performance-settings',
+				label: __( 'Performance settings' ),
+				callback: ( sites ) => {
+					const site = sites[ 0 ];
+					const wpAdminUrl = getSiteAdminUrl( site );
+					const adminInterface = getAdminInterface( site );
+					const isWpAdminInterface = adminInterface === 'wp-admin';
+					window.location.href = isWpAdminInterface
+						? `${ wpAdminUrl }options-general.php?page=page-optimize`
+						: `/settings/performance/${ site.slug }`;
+					dispatch(
+						recordTracksEvent( 'calypso_sites_dashboard_site_action_performance_settings_click' )
+					);
+				},
+				isEligible: ( site ) => {
+					const adminInterface = getAdminInterface( site );
+					const isWpAdminInterface = adminInterface === 'wp-admin';
+					const isClassicSimple = isWpAdminInterface && isSimpleSite( site );
+					return ! isClassicSimple;
+				},
+			},
+
+			{
+				id: 'privacy-settings',
+				label: __( 'Privacy settings' ),
+				callback: ( sites ) => {
+					const site = sites[ 0 ];
+					window.location.href = `/settings/general/${ site.slug }#site-privacy-settings`;
+					dispatch(
+						recordTracksEvent( 'calypso_sites_dashboard_site_action_privacy_settings_click' )
+					);
+				},
+				isEligible: ( site ) => {
+					const isLaunched = site.launch_status !== 'unlaunched';
+					return isLaunched;
+				},
+			},
+
+			{
+				id: 'domains-and-dns',
+				label: __( 'Domains and DNS' ),
+				callback: ( sites ) => {
+					const site = sites[ 0 ];
+					window.location.href = `/domains/manage/${ site.slug }/dns/${ site.slug }`;
+					dispatch(
+						recordTracksEvent( 'calypso_sites_dashboard_site_action_domains_and_dns_click' )
+					);
+				},
+				isEligible: ( site ) => {
+					const hasCustomDomain = isCustomDomain( site.slug );
+					const isSiteJetpackNotAtomic = isNotAtomicJetpack( site );
+					return hasCustomDomain && ! isSiteJetpackNotAtomic;
+				},
+			},
+
+			{
+				id: 'admin',
+				label: __( 'WP Admin' ),
+				callback: ( sites ) => {
+					const site = sites[ 0 ];
+					window.location.href = site.options?.admin_url ?? '';
```

Comment:
I don't understand? We're navigating to the admin here, we're not redirecting to self?

Reply:
Ah. I was talking about the `?? ''` case, which would happen if the `admin_url` option is not set (which may be rare, I presume)

Reply:
Yeah to be honest I'm not sure, I only added that because of a typescript error. I think I'll add an `if` condition it's a bit better.

Reply:
It's pretty safe to assume that a site always has an `admin_url`. The default is being added only to make the typechecker happy. The `site.options` are also always there.

If you truly want to avoid navigating to a `null` URL, you can make the navigation conditional.
```js
if ( site.options.admin_url ) { navigateTo( site.options.admin_url ); }
```
but there's very little practical benefit.